File: examples/data-processing.js
#!/usr/bin/env node
/**
 * Data Processing Example in JavaScript
 * Demonstrates working with arrays, objects, and JSON
 */

// Process an array of numbers
const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
console.log(`Original numbers: ${JSON.stringify(numbers)}`);

// Calculate squares
const squared = numbers.map(n => n ** 2);
console.log(`Squared: ${JSON.stringify(squared)}`);

// Calculate sum
const total = squared.reduce((acc, n) => acc + n, 0);
console.log(`Sum of squares: ${total}`);

// Filter even numbers
const evens = numbers.filter(n => n % 2 === 0);
console.log(`Even numbers: ${JSON.stringify(evens)}`);

// Create a result object
const result = {
  status: "success",
  count: numbers.length,
  sum_of_squares: total,
  even_numbers: evens,
  statistics: {
    min: Math.min(...numbers),
    max: Math.max(...numbers),
    avg: numbers.reduce((a, b) => a + b, 0) / numbers.length
  }
};

// Output as JSON
console.log("\nResult as JSON:");
console.log(JSON.stringify(result, null, 2));

File: examples/data-processing.py
#!/usr/bin/env python3
"""
Data Processing Example in Python
Demonstrates working with lists, dictionaries, and JSON
"""

import json

# Process a list of numbers
numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print(f"Original numbers: {numbers}")

# Calculate squares
squared = [n**2 for n in numbers]
print(f"Squared: {squared}")

# Calculate sum
total = sum(squared)
print(f"Sum of squares: {total}")

# Filter even numbers
evens = [n for n in numbers if n % 2 == 0]
print(f"Even numbers: {evens}")

# Create a result object
result = {
    "status": "success",
    "count": len(numbers),
    "sum_of_squares": total,
    "even_numbers": evens,
    "statistics": {
        "min": min(numbers),
        "max": max(numbers),
        "avg": sum(numbers) / len(numbers)
    }
}

# Output as JSON
print("\nResult as JSON:")
print(json.dumps(result, indent=2))
